sklearn/utils/estimator_checks.py

```diff
@@ -1749,3 +1783,215 @@ def check_decision_proba_consistency(name, estimator_orig):
         a = estimator.predict_proba(X_test)[:, 1]
         b = estimator.decision_function(X_test)
         assert_array_equal(rankdata(a), rankdata(b))
+
+
+@ignore_warnings(category=(DeprecationWarning))
+def check_classes_parameter_predict_proba(name, estimator_orig):
+    # Test whether adding extra classes doesn't change the prediction of the
+    # existing classes just adds 0 value columns for new classes
+    rng = np.random.RandomState(0)
+    X = rng.randint(5, size=(6, 100))
+    y = np.array([1, 1, 1, 2, 2, 2])
+    params = {}
+
+    clf = clone(estimator_orig)
+    pred0 = clf.fit(X, y).predict_proba(X)
+
+    params['classes'] = [1, 2]
+    clf.set_params(**params)
```

Comment:
This requires the estimator to have a classes attribute and in turn creates a backwards incompatibility for users of `check_estimator`. At least for a couple of versions, we should only warn if a classifier providing `predict_proba`, `predict_log_proba` or `decision_function` lacks a `classes` parameter.

Reply:
right now, these checks are only for the classifiers which i've updated. but I think you're talking about user-defined estimators. would it work that we should just perform this check if classes is present as a parameter in the estimator? will we be specifying that classes parameter should be present in all user-defined classifiers?

Reply:
I'm actually not sure if we should require that. It's a good question. I think for now it would be fine if we skip the test if ``classes`` is not a parameter.